import/WebView.qml

```diff
@@ -0,0 +1,264 @@
+                    sendAsyncMessage("embedui:premissions",
+                                     { "allow": false, "checkedDontAsk": false, "id": data.id })
+                }
+                break
+            }
+            case "embed:login": {
+                dialog = pageStack.push(passwordManagerDialog,
+                                        { "webView": webView, "requestId": data.id,
+                                          "notificationType": data.name, "formData": data.formdata })
+                break
+            }
+            case "Content:ContextMenu": {
+                if (data.types.indexOf("image") !== -1 || data.types.indexOf("link") !== -1) {
+                    var linkHref = data.linkURL
+                    var imageSrc = data.mediaURL
+                    var linkTitle = data.linkTitle
+                    var contentType = data.contentType
+                    if (privData.contextMenu) {
+                        privData.contextMenu.linkHref = linkHref
+                        privData.contextMenu.imageSrc = imageSrc
+                        privData.contextMenu.linkTitle = linkTitle.trim()
+                        privData.contextMenu.linkProtocol = data.linkProtocol || ""
+                        privData.contextMenu.contentType = contentType
+                        privData.contextMenu.viewId = privData.uniqueId()
+                        privData.contextMenu.tabModel = null // TODO ??
+                        // PageStack and WebView are currently always the same but
+                        // let's update them regardless so that they will remain correct.
+                        privData.contextMenu.pageStack = pageStack
+                        privData.hideVirtualKeyboard()
+                        privData.contextMenu.show()
+                    } else {
+                        privData.contextMenu = contextMenuComponent.createObject(webView,
+                                { "linkHref": linkHref, "imageSrc": imageSrc,
+                                  "linkTitle": linkTitle.trim(), "linkProtocol": data.linkProtocol,
+                                  "contentType": contentType, "tabModel": null,
+                                  "viewId": privData.uniqueId(),
+                                  "pageStack": pageStack })
+                        privData.hideVirtualKeyboard()
+                        privData.contextMenu.show()
+                    }
+                }
+                break
+            }
+            default: {
+                console.log("unhandled async message received: " + message)
+                break
+            }
+        }
+    }
+
+    BusyIndicator {
+        id: busySpinner
+        anchors.centerIn: parent
+        running: true
+        visible: webview.loading
+        size: BusyIndicatorSize.Large
+    }
+
+    Timer {
+        id: helper
+        property var geolocationUrls: {{}}
+        property var contextMenu
+
+        function uniqueId() {
```

Comment:
This should be webview's uniqueId not just a random id. However, with download manager I checked quickly and it looks that viewId is no longer used.